builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -326,10 +334,37 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 				ForceNew: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"template": &schema.Schema{
-							Type:     schema.TypeString,
+						"clone": &schema.Schema{
+							Type:     schema.TypeMap,
 							Optional: true,
 							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"label": &schema.Schema{
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+									"linked": &schema.Schema{
+										Type:     schema.TypeBool,
+										Optional: true,
+										Default:  false,
+										ForceNew: true,
+									},
+									"snapshot": &schema.Schema{
+										Type:     schema.TypeString,
+										Optional: true,
+										ForceNew: true,
```

Comment:
These can hold any value (string, bool and string).

We do make the appropriate tests in the code itself. For instance, you cannot clone from a snapshot that doesn't exist, etc.